source/Calamari.Shared/Integration/Packages/CombinedPackageExtractor.cs

```diff
@@ -13,37 +13,33 @@ public interface ICombinedPackageExtractor : IPackageExtractor
 
     public class CombinedPackageExtractor : ICombinedPackageExtractor
     {
-        readonly ISupportLinkGenerator supportLinkGenerator = new SupportLinkGenerator();
-
+        /// Order is important here since .tar.gz should be checked for before .gz
+        static IList<IPackageExtractor> Extractors => new List<IPackageExtractor>
+        {
+            new NupkgExtractor(),
+            new TarGzipPackageExtractor(),
+            new TarBzipPackageExtractor(),
+            new ZipPackageExtractor(),
+            new TarPackageExtractor()
+        };
+        
         public string[] Extensions => Extractors.SelectMany(e => e.Extensions).OrderBy(e => e).ToArray();
 
         public int Extract(string packageFile, string directory, bool suppressNestedScriptWarning)
-        {
-            return GetExtractor(packageFile).Extract(packageFile, directory, suppressNestedScriptWarning);
-        }
+            => GetExtractor(packageFile).Extract(packageFile, directory, suppressNestedScriptWarning);
 
-        public IPackageExtractor GetExtractor(string packageFile)
+        public static IPackageExtractor GetExtractor(string packageFile)
         {
             var extension = Path.GetExtension(packageFile);
             if (string.IsNullOrEmpty(extension))
-            {
-                throw new FileFormatException("Package is missing file extension. This is needed to select the correct extraction algorithm.");
-            }
+                throw new CommandException("Package is missing file extension. This is needed to select the correct extraction algorithm.");
 
             var file = PackageName.FromFile(packageFile);
-            if (!Extensions.Contains(file.Extension))
-            {
-                throw new FileFormatException($"Unsupported file extension `{extension}`");
-            }
-
-            var extractor = FindByExtension(file);
-            if (extractor != null)
-                return extractor;
+            var extractor = Extractors.FirstOrDefault(p => p.Extensions.Any(ext => file.Extension.Equals(ext, StringComparison.OrdinalIgnoreCase)));
+            if (extractor == null)
+                throw new CommandException($"Unsupported file extension `{extension}`");
             
-            throw new FileFormatException(supportLinkGenerator.GenerateSupportMessage(
```

Comment:
I might be reading this wrong, but `extractor` should never be null as line `34` above already checked the extension exists. Therefore this error was never thrown, so might as well remove it.